tutoroars/templates/oars/jobs/init/clickhouse/oars_init_schemas_tables_users.sh

```diff
@@ -84,9 +84,8 @@ CREATE TABLE IF NOT EXISTS {{ OARS_XAPI_DATABASE }}.{{ OARS_ENROLLMENT_EVENTS_TA
     `object_id` String NOT NULL,
     `course_id` String NOT NULL,
     `org` String NOT NULL,
-    `verb_id` String NOT NULL,
-    `event_type` String NOT NULL,
-    `enrollment_mode` String
+    `verb_id` LowCardinality(String) NOT NULL,
+    `enrollment_mode` LowCardinality(String)
 ) ENGINE = MergeTree
 PRIMARY KEY (org, course_id, enrollment_mode)
 ORDER BY (org, course_id, enrollment_mode, emission_time);
```

Comment:
Unrelated to this PR, but is there a reason why we're not using the actor id in the key / order?

Reply:
My reason for leaving it out of the key is that the recommendation I've seen from the ClickHouse docs is to prefer sparser primary keys, and I felt that actor_id might have a higher cardinality than is preferred for primary keys. However, that's just an assumption on my part and I don't have benchmarks behind that one way or the other.

That said, I don't have a reason for leaving it out entirely and I think it should be added. I have a slight preference for including it in the ordering, between `enrollment_mode` and `emission_time`, but if you have a preference for the key then I'd say let's go with that.

Reply:
I think it should be included since IIRC we're going to be grouping on it a lot to find the most recent state? I'm ok with it in that location. We can do some optimization tests later on the actual query patterns are established.

Reply:
That makes sense. I'll include it in the other tables as well.

Reply:
Ok, sorry, one last comment: I think you're right and that since the expected query pattern is to get the most recent state, `actor_id` should be ahead of `enrollment_mode`. In that case, I propose the following:
```sql
primary key (org, course_id)
order by (org, course_id, actor_id, enrollment_mode, emission_time)
```
What do you think?

Reply:
Seems legit, let's try it!